### PR TITLE
add self assessment for security

### PR DESCRIPTION
**Pull request by vincentgoat**
Signed-off-by: vincentgoat <linguohui1@huawei.com>

**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Adding self assessment for security, refer to the template of cncf/tag-security [here](https://github.com/cncf/tag-security/blob/main/assessments/guide/self-assessment.md)

cc @fisherxu @kevin-wangzefeng 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: docs/security/self-assessment.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+This document serves to provide KubeEdge users with an initial understanding of KubeEdge's security, where to find existing security documentation, KubeEdge plans for security, and general overview of KubeEdge security practices, both for development of KubeEdge as well as security of KubeEdge.
+
+This document provides the CNCF TAG-Security with an initial understanding of KubeEdge to assist in a joint-review, necessary for projects under incubation.  Taken together, this document and the joint-review serve as a cornerstone for if and when KubeEdge seeks graduation and is preparing for a security audit.
+
+## Security functions and features
+
+* Critical.  The cloud component CloudHub exposes ports to provide external services, accept connections from edge nodes, and be responsible for cloud-side communication. During communication, webSocket or Quick protocol is used for data transmission, and TLS certificate is used for data encryption.
+* Critical.  The EdgeHub component exposes the port to provide services externally, and provides list&Watch capability externally on the edge side.
```
Comment (kevin-wangzefeng) [inline docs/security/self-assessment.md:108]: This paragraph is confusing, I think we should list a set of security relevant functions and features. E.g. what we did to protect from security threat.

File: docs/security/self-assessment.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+All code is maintained in [Git](https://github.com/kubeedge/kubeedge) and changes must be reviewed by maintainers and must pass all unit and e2e tests. Code changes are submitted via Pull Requests (PRs) and must be signed. Commits to the `main` branch directly are not allowed.
+
+### Communication Channels
+
+* Internal. [GitHub Discussions](https://github.com/kubeedge/kubeedge/discussions)
+* Inbound. How do users or prospective users communicate with the team?
+
+- Inbound: [Slack Channel](https://join.slack.com/t/kubeedge/shared_invite/enQtNjc0MTg2NTg2MTk0LWJmOTBmOGRkZWNhMTVkNGU1ZjkwNDY4MTY4YTAwNDAyMjRkMjdlMjIzYmMxODY1NGZjYzc4MWM5YmIxZjU1ZDI), [mailing list](https://groups.google.com/forum/#!forum/kubeedge)
+- Outbound: [Slack Channel](https://join.slack.com/t/kubeedge/shared_invite/enQtNjc0MTg2NTg2MTk0LWJmOTBmOGRkZWNhMTVkNGU1ZjkwNDY4MTY4YTAwNDAyMjRkMjdlMjIzYmMxODY1NGZjYzc4MWM5YmIxZjU1ZDI), [mailing list](https://groups.google.com/forum/#!forum/kubeedge)
+- Security email group: [kubeedge-security](mailto:cncf-kubeedge-security@lists.cncf.io), [distributors announce list](mailto:cncf-kubeedge-distrib-announce@lists.cncf.io)
```
Comment (kevin-wangzefeng) [inline docs/security/self-assessment.md:129]: You may keep the original question body and add answer in new lines.

File: docs/security/self-assessment.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+KubeEdge has achieved the passing level criteria for [CII Best Practices](https://bestpractices.coreinfrastructure.org/en/projects/3018).
+
+### Development Pipeline
+
+All code is maintained in [Git](https://github.com/kubeedge/kubeedge) and changes must be reviewed by maintainers and must pass all unit and e2e tests. Code changes are submitted via Pull Requests (PRs) and must be signed. Commits to the `main` branch directly are not allowed.
+
+### Communication Channels
+
+* Internal. [GitHub Discussions](https://github.com/kubeedge/kubeedge/discussions)
```
Comment (kevin-wangzefeng) [inline docs/security/self-assessment.md:124]: We currently don't use github discussions that much. Most communications happen through community meeting, slack, and github issues.

Comment (kevin-wangzefeng) [reply docs/security/self-assessment.md:124]: Community meeting for example, we have regular meeting alternatve between Europe friendly time and Pacific friendly time. Besides, we have SIG meetings and on demond meetings. All are publicly accessable and meeting records are uploaded to youtube.

File: docs/security/self-assessment.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+* Inbound. How do users or prospective users communicate with the team?
+
+- Inbound: [Slack Channel](https://join.slack.com/t/kubeedge/shared_invite/enQtNjc0MTg2NTg2MTk0LWJmOTBmOGRkZWNhMTVkNGU1ZjkwNDY4MTY4YTAwNDAyMjRkMjdlMjIzYmMxODY1NGZjYzc4MWM5YmIxZjU1ZDI), [mailing list](https://groups.google.com/forum/#!forum/kubeedge)
+- Outbound: [Slack Channel](https://join.slack.com/t/kubeedge/shared_invite/enQtNjc0MTg2NTg2MTk0LWJmOTBmOGRkZWNhMTVkNGU1ZjkwNDY4MTY4YTAwNDAyMjRkMjdlMjIzYmMxODY1NGZjYzc4MWM5YmIxZjU1ZDI), [mailing list](https://groups.google.com/forum/#!forum/kubeedge)
+- Security email group: [kubeedge-security](mailto:cncf-kubeedge-security@lists.cncf.io), [distributors announce list](mailto:cncf-kubeedge-distrib-announce@lists.cncf.io)
+
+### Ecosystem
+
+KubeEdge is based on Kubernetes base, 100% compatible with Kubernetes, support declarative API, CRD, scalability. Build cloud-side consistent containers and Kubernetes ecology, application portability. Cloud capabilities are extended to the edge through edge cloud collaboration, including AI collaboration, data collaboration, application collaboration and management collaboration.
```
Comment (kevin-wangzefeng) [inline docs/security/self-assessment.md:133]: Not only integration and compatibility with Kubernetes, but also include how KubeEdge is exploring the frontiers of Cloud Native, and helping more users from different industry to adpot cloud native technologies.

File: docs/security/self-assessment.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+* e Cloud of China Telecom International Co., Ltd., has built a large-scale CDN node management platform base on KubeEdge. Cloud-side collaboration and edge autonomy are provided by KubEedge, while most of the original capabilities of K8S are retained.
```
Comment (kevin-wangzefeng) [inline docs/security/self-assessment.md:163]:
```suggestion

* e Cloud of China Telecom International Co., Ltd., has built a large-scale CDN node management platform base on KubeEdge. Cloud-side collaboration and edge autonomy are provided by KubEedge, while most of the original capabilities of K8S are retained.
```

File: docs/security/self-assessment.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+| Website           | https://kubeedge.io                                          |
+| Security Provider | No                                                           |
+| Languages         | Go                                                           |
+| SBOM              | Check [go.mod](https://github.com/kubeedge/kubeedge/blob/master/go.mod) for libraries, packages, versions used by the project |
+
+### Security links
+
+| Doc                          | url                                                          |
+| ---------------------------- | ------------------------------------------------------------ |
+| Security file                | [Security policy](https://github.com/kubeedge/kubeedge/security/policy) |
```
Comment (kevin-wangzefeng) [inline docs/security/self-assessment.md:35]:
```suggestion
| Security file                | [Security policy](https://github.com/kubeedge/kubeedge/blob/master/SECURITY.md) |
```

File: docs/security/self-assessment.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+| Default and optional configs | [cloudcore](https://github.com/kubeedge/kubeedge/blob/master/pkg/apis/componentconfig/cloudcore/v1alpha1/default.go)<br />[edgecore](https://github.com/kubeedge/kubeedge/blob/master/pkg/apis/componentconfig/edgecore/v1alpha1/default.go) |
+
+## Overview
```
Comment (kevin-wangzefeng) [inline docs/security/self-assessment.md:38]: Some basic information seems out of date, please check whether the orginal reference is up to or out of date.

File: docs/security/self-assessment.md [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+
+### Background
+
+With the rapid development of cloud, 5G and AI technologies, enterprises are increasingly demanding intelligent upgrading, and the application scenarios of edge computing are becoming more and more extensive. It is challenging to realize unified management and control of edge computing resources and cloud-edge synergy, such as how to run intelligent applications and algorithms on edge devices with limited resources(e.g. cameras and drones), how to solve the problems caused by the access of massive heterogeneous edge devices in intelligent transportation and intelligent power scenarios, and how to ensure high reliability of services in off-line scenarios.
```
Comment (vincentgoat) [inline docs/security/self-assessment.md:46]: [FYI](http://www.ecconsortium.net/Lists/show/id/518.html)

File: docs/security/self-assessment.md [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+
+
+* In China’s highway electronic toll collection (ETC) system, KubeEdge helps manage nearly 100,000 edge nodes and more than 500,000 edge applications in 29 of China’s 34 provinces, cities and autonomous regions. With these applications, the system processes more than 300 million data records daily and has improved traffic efficiency at the toll stations by 10 times.
+* China Mobile On-line Marketing Service Center, is a secondary organ of the China Mobile Communications Group, holds the world’s largest call center with 44,000 agents, 900 million users, and 53 customer service centers, builds a cloud-edge synergy architecture consisting of two centers and multiple edges based on KubeEdge. See more details [here](https://www.cncf.io/blog/2021/08/16/china-mobile-kubeedge-based-customer-service-platform-featuring-edge-cloud-synergy).
```
Comment (vincentgoat) [inline docs/security/self-assessment.md:181]: [FYI](https://vmblog.com/archive/2021/04/16/managing-100-000-edge-nodes-on-china-s-highways-using-kubeedge.aspx#.YmfoS-dByUk),[FYI](https://www.cncf.io/blog/2021/01/25/maintainer-spotlight-kevin-wang-of-kubeedge-and-volcano/)

**Comment by kevin-wangzefeng** [comment]
The DCO check is failing on your last commit.

**Comment by vincentgoat** [comment]
Instead by https://github.com/kubeedge/community/pull/81

**Pull request closed without merging**